File: spring-aws-s3-java/src/main/java/com/migangqui/spring/aws/s3/service/AmazonS3Service.java
package com.migangqui.spring.aws.s3.service;

import com.migangqui.spring.aws.s3.bean.UploadFileResult;

import java.io.InputStream;

public interface AmazonS3Service {
	
	UploadFileResult uploadFile(InputStream stream, String folder, String name, String contentType);

	UploadFileResult uploadFile(byte[] bytes, String folder, String name, String contentType);

	UploadFileResult uploadFileAsync(InputStream stream, String folder, String name, String contentType);

	UploadFileResult uploadFileAsync(byte[] bytes, String folder, String name, String contentType);
	
	InputStream getFile(String path);
	
	boolean deleteFile(String path);

}

File: spring-aws-s3-kotlin/src/main/kotlin/com/migangqui/spring/aws/s3/service/AmazonS3Service.kt
package com.migangqui.spring.aws.s3.service

import com.migangqui.spring.aws.s3.bean.UploadFileResult
import java.io.InputStream

interface AmazonS3Service {

    fun uploadFile(stream: InputStream, folder: String, name: String, contentType: String): UploadFileResult

    fun uploadFile(bytes: ByteArray, folder: String, name: String, contentType: String): UploadFileResult

    fun uploadFileAsync(stream: InputStream, folder: String, name: String, contentType: String): UploadFileResult

    fun uploadFileAsync(bytes: ByteArray, folder: String, name: String, contentType: String): UploadFileResult

    fun getFile(path: String): InputStream

    fun deleteFile(path: String): Boolean
}